src/tests/frontend/specs/adminsettings.js

```diff
@@ -3,6 +3,19 @@
 describe('Admin > Settings', function () {
   this.timeout(480000);
 
+  let saveProgressReceived;
+  let observer;
+  // observes the #response element' attributes for changes
+  const observerJob = () => {
+    saveProgressReceived = false;
+    const observerCallback = (mutations, observer) => {
+      saveProgressReceived = true;
+    };
+    observer = new MutationObserver(observerCallback);
+    observer.observe(helper.admin$('#response')[0],
+        {attributes: true, childList: false, subtree: false});
+  };
```

Comment:
Rather than expose the observer, use a self-contained function:

```javascript
    const save = async () => {
      const p = new Promise((resolve) => {
        const observer = new MutationObserver(() => { resolve(); observer.disconnect(); });
        observer.observe(
            helper.admin$('#response')[0], {attributes: true, childList: false, subtree: false});
      });
      helper.admin$('#saveSettings').click();
      await p;
    };
```